client/proxy/proxy.go

```diff
@@ -34,76 +33,23 @@ type Client interface {
 
 // NewManagerClient creates a new ManagerClient instance. It trims the apiVersion of leading and trailing slashes
 // and the base path of trailing slashes to ensure consistency
-func NewManagerClient(base url.URL, apiVersion string, client *http.Client) *ManagerClient {
-	base.Path = strings.TrimSuffix(base.Path, "/")
-	return &ManagerClient{
-		base:             base,
-		versionedAPIPath: strings.TrimPrefix(strings.TrimSuffix(apiVersion, "/"), "/"),
-		client:           client,
-	}
-}
-
-func (m *ManagerClient) toCurl(request *http.Request, body string) string {
-
```

Comment:
Oh nice. Then the extra curl kludge is unnecessary for k8s, but needed in generic mode.